hubble: bump proto deps for cmd/protoc-gen-go-grpc/v1.1.0

rolinh opened the pull request.

rolinh committed:
image/hubble-proto: bump deps for cmd/protoc-gen-go-grpc/v1.1.0

See release notes:
https://github.com/grpc/grpc-go/releases/tag/cmd%2Fprotoc-gen-go-grpc%2Fv1.0.1
https://github.com/grpc/grpc-go/releases/tag/cmd%2Fprotoc-gen-go-grpc%2Fv1.1.0

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

rolinh committed:
api/v1: re-gen proto for proto deps updates

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

glibsm commented on images/hubble-proto/install-protoplugins.sh:
```diff
@@ -14,11 +14,12 @@ go mod init github.com/cilium/hubble/protoc
 # work with nested go.mod definitions.
-go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@cee815d
+# b2c5f4a == tag: cmd/protoc-gen-go-grpc/v1.0.0
+go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@b2c5f4a
```
Why are we not using `@1.0.0` here?

rolinh replied:
The label is literally `cmd/protoc-gen-go-grpc/v1.0.0`, not `v1.0.0` (which would point to an older release of `grpc-go`, the most recent one being `1.32.0`). And this format does not work with `go get`:
```
$ go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@cmd/protoc-gen-go-grpc/v1.0.0: google.golang.org/grpc/cmd/protoc-gen-go-grpc@cmd/protoc-gen-go-grpc/v1.0.0: invalid version: version "cmd/protoc-gen-go-grpc/v1.0.0" invalid: disallowed version string
```

glibsm replied:
> The label is literally cmd/protoc-gen-go-grpc/v1.0.0

D:

glibsm commented on images/hubble-proto/install-protoplugins.sh:
```diff
@@ -14,11 +14,12 @@ go mod init github.com/cilium/hubble/protoc
+go get github.com/mitchellh/protoc-gen-go-json@364b693
 go build github.com/mitchellh/protoc-gen-go-json
 
 go get github.com/golang/protobuf@v1.4.2
```
may need to double check this version is compatible with `3.13.0`

rolinh committed:
image/hubble-proto: ensure the container can be run as a regular user

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

rolinh committed:
api/v1: run hubble-proto container as a regular user

When run as root, the owner of newly generated files ends up being root.
Run the container as a regular user to fix this issue.

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

glibsm commented on images/hubble-proto/Makefile.docker:
```diff
@@ -6,6 +6,7 @@
 
 PROTOC ?= protoc
 
+EXTERNAL_PROTO_SOURCE := /usr/local/include
```
this path should be auto-inlcuded, no?

rolinh replied:
It doesn't seem to be the case when running as a regular user.

tklauser commented on images/hubble-proto/install-protoplugins.sh:
```diff
@@ -14,12 +14,13 @@ go mod init github.com/cilium/hubble/protoc
 # work with nested go.mod definitions.
-go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@cee815d
+# a80e650 == tag: cmd/protoc-gen-go-grpc/v1.0.1
+go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@a80e65018272ac6bd58ee44d0d3e37374b090c1f
```
Cannot comment on the lines above, but dropping the `unset GOPATH` and `go mod init` and instead setting `GO111MODULE=on` (which will be the default in the upcoming Go 1.16) and `GOPATH=/proto` would allow to use single line `go get ...` with proper version tags to get and build all the binaries. However, they will be installed to `$GOPATH/bin` rather than `/proto` (haven't found a way to override this behavior), so the `COPY` in `Dockerfile` would need to be adjusted as well.

Something like the following:

```bash
#!/bin/bash

set -o xtrace
set -o errexit
set -o pipefail
set -o nounset

mkdir /proto

export GOPATH=/proto
export GO111MODULE=on

go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.0.1

# protoc-gen-go-json doesn't have releases, this is the latest commit at the time
go get github.com/mitchellh/protoc-gen-go-json@364b693

go get github.com/golang/protobuf@v1.4.3
```

and the diff for `Dockerfile`:

```diff
--- a/images/hubble-proto/Dockerfile
+++ b/images/hubble-proto/Dockerfile
@@ -33,9 +33,9 @@ RUN /tmp/install-glibc.sh
 COPY install-protoc.sh /tmp/install-protoc.sh
 RUN /tmp/install-protoc.sh
 
-COPY --from=builder /proto/protoc-gen-go .
-COPY --from=builder /proto/protoc-gen-go-grpc .
-COPY --from=builder /proto/protoc-gen-go-json .
+COPY --from=builder /proto/bin/protoc-gen-go .
+COPY --from=builder /proto/bin/protoc-gen-go-grpc .
+COPY --from=builder /proto/bin/protoc-gen-go-json .
```

rolinh replied:
Thanks. Just tried it but it's somehow not working (I didn't investigate further, not sure it's worth the trouble).

tklauser replied:
Certainly not worth the trouble and if you want I can try to take this up in a follow-up PR once this one is merged.

rolinh committed:
vendor: bump etcd to v3.5.0-alpha.0 and grpc to v1.36.0

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>
